receiver/receivercreator/rules.go

```diff
@@ -30,7 +30,7 @@ type rule struct {
 }
 
 // ruleRe is used to verify the rule starts type check.
-var ruleRe = regexp.MustCompile(`^type\s*==\s*("pod"|"port"|"hostport")`)
+var ruleRe = regexp.MustCompile(`^type\s*==\s*("pod"|"port"|"hostport"|"container")`)
```

Comment:
It may be preferable to share symbols from `observer`:

```suggestion
var ruleRe = regexp.MustCompile(
	fmt.Sprintf(`^type\s*==\s*(%q|%q|%q|%q)`, observer.PodType, observer.PortType, observer.HostPortType, observer.ContainerType),
)
```